src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -724,6 +858,34 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
         // here due to a typo
         return this.laserPointerDefaultColor;
     }
+
+    private _pickWithMesh(handMesh: AbstractMesh, precise: boolean, predicate: (mesh: AbstractMesh) => boolean): Nullable<PickingInfo> {
+        var pickingInfo = new PickingInfo();
+        if (handMesh) {
+            for (let meshIndex = 0; meshIndex < this._scene.meshes.length; meshIndex++) {
+                let mesh = this._scene.meshes[meshIndex];
+                if (!predicate(mesh)) {
+                        continue;
+                    }
+                let result = mesh.intersectsMesh(handMesh, precise);
+                if (result) {
+                    pickingInfo.hit = result;
+                    pickingInfo.pickedMesh = mesh;
+                    pickingInfo.pickedPoint = handMesh.position;
+                    pickingInfo.originMesh = handMesh;
+                    pickingInfo.distance = Vector3.Distance(mesh.position, handMesh.position);
+                    mesh.updateFacetData();
+                    const closestFacet = mesh.getClosestFacetAtCoordinates(handMesh.position.x, handMesh.position.y, handMesh.position.z, undefined, true, true);
+                    if (closestFacet)
+                    {
+                        pickingInfo.ray = new Ray(handMesh.position, mesh.getFacetNormal(closestFacet).negate(), pickingInfo.distance);

```

Comment:
Is it possible to reuse this ray instead of creating a new one?  This happens on each frame and will have impact on the GC.

Same goes to the `getFacetNormal` call (use `getFacetNormalToRef` instead, and negate in place, so you don't create new vectors)

Reply:
I would love to do that, but it changes for every frame right? I create the ray depending on the index finger position which can change every frame.

Reply:
The controller data object has a tmpRay that is reused on each frame. You can use it (if it makes sense) or create a new one for this interaction.